x/axelarnet/types/evm_translator.go

```diff
@@ -190,25 +190,34 @@ func ConstructWasmMessageV2(gm nexus.GeneralMessage, payload []byte) ([]byte, er
 		}
 	}
 
-	msg := wasm{
+	// To avoid unintended transformations of the payload due to decoding to a map[string]interface{}
+	// persisting to the outgoing payload (such as number to float conversion), we wrap the metadata over it
```

Comment:
```suggestion
	// re-marshalling the json payload can lead to subtle changes (e.g. number to float conversion + rounding errors),
	// so we leave the payload blank before marshalling and then inject the original payload into the json string instead
```

Reply:
sure

Reply:
more context is fine. I just know that I won't understand what you mean with the original comment when I come back to this code in a few weeks